test/test_torch.py

```diff
@@ -4713,6 +4713,26 @@ def run_test(M, sign=1):
     def test_matrix_power(self):
         self._test_matrix_power(self, conv_fn=lambda x: x)
 
+    @staticmethod
+    def _test_chain_matmul(self, cast):
+        def product(matrices):
+            for mat in matrices[1:]:
+                matrices[0] = matrices[0].mm(mat)
+            return matrices[0]
+
+        def run_test(p, cast):
+            matrices = []
+            for (pi, pi_1) in zip(p[:-1], p[1:]):
+                matrices.append(cast(torch.randn(pi, pi_1)))
+            self.assertEqual(torch.chain_matmul(*matrices), product(matrices))
+
+        run_test([10, 20, 30, 5], cast)
```

Comment:
add a test for 3 matrices since you specialized the impl.

Reply:
Length of `p` is N + 1 where N is the number of matrices. N=3 is covered.

Reply:
Ah yes you are right!